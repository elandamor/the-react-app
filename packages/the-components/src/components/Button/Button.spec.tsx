import { ThemeProvider } from '@elandamor/tra-theme';
import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Button, IButtonProps } from './index';

const setup = (props?: IButtonProps) => {
  return render(
    <ThemeProvider>
      <Button {...props} />
    </ThemeProvider>,
  );
};

describe('Button', () => {
  it('renders without crashing', () => {
    const { container } = setup();
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('button');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a text button', () => {
    const { getByText } = setup();
    expect(getByText('Button')).toBeTruthy();
  });

  it('handles onClick event', () => {
    function Counter() {
      const [count, setCount] = React.useState(0);

      const props = {
        onClick: () => setCount(count + 1),
        text: count.toString(),
      };

      return <Button data-testid="button" {...props} />;
    }

    const { getByTestId } = render(
      <ThemeProvider>
        <Counter />
      </ThemeProvider>,
    );

    const buttonNode = getByTestId('button');
    fireEvent.click(buttonNode);

    expect(buttonNode).toHaveTextContent('1');
  });
});
