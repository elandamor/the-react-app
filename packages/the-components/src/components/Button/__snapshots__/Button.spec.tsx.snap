// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.0892857143em;
  -moz-letter-spacing: 0.0892857143em;
  -ms-letter-spacing: 0.0892857143em;
  letter-spacing: 0.0892857143em;
  outline: none;
  overflow: hidden;
  padding: 8px;
  position: relative;
  text-transform: uppercase;
  z-index: 1;
}

.emotion-1:after,
.emotion-1:before {
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 0;
}

.emotion-1:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.emotion-1[disabled] {
  opacity: 0.5;
}

.emotion-1:not([disabled]):focus:before {
  background-color: rgba(0,0,0,.1);
}

.emotion-1:not([disabled]):hover:before {
  background-color: rgba(0,0,0,.05);
}

.emotion-0 {
  color: inherit;
  line-height: 1.5;
  line-height: 1;
}

<button
  class="emotion-1"
>
  <span
    class="emotion-0"
  >
    Button
  </span>
</button>
`;
